relay/relay_test.go

```diff
@@ -521,15 +520,27 @@ func (t *testRelaySuite) TestReSetupMeta(c *C) {
 	uuid002 := fmt.Sprintf("%s.000002", uuid)
 	t.verifyMetadata(c, r, uuid002, minCheckpoint, r.cfg.BinlogGTID, []string{uuid001, uuid002})
 
+	r.cfg.BinLogName = "mysql-bin.000002"
+	r.cfg.BinlogGTID = "24ecd093-8cec-11e9-aa0d-0242ac170002:1-50,24ecd093-8cec-11e9-aa0d-0242ac170003:1-50"
+	r.cfg.UUIDSuffix = "000002"
+	c.Assert(r.reSetupMeta(ctx), IsNil)
+	t.verifyMetadata(c, r, uuid002, gmysql.Position{Name: r.cfg.BinLogName, Pos: 4}, r.cfg.BinlogGTID, []string{uuid002})
+
+	// re-setup meta again, often happen when connecting a server behind a VIP.
```

Comment:
What's VIP 🤔

Reply:
virtual IP address for master and slave. It can automatic switch master-slave in case of failure.